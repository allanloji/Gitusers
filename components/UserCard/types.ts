type UserCardProps = {
  avatarUrl: string;
  login: string;
  type: string;
};

export type { UserCardProps };
