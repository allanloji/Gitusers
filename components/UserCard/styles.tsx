import styled from '@emotion/styled';

const Container = styled.div`
  background-color: rgb(43, 43, 43);
  display: flex;
  //padding: 1.5rem;
  border-radius: 5px;
  box-shadow: 5px 5px 15px 5px rgba(0, 0, 0, 0.05);
  //align-items: center;
  //justify-content: space-evenly;

  //width: 100px;
`;

const AvatarContainer = styled.div<{ color: string }>`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${({ color }) => color};
  border-radius: 5px;
  width: 100px;
`;

const Avatar = styled.img`
  width: 80px;
  height: 80px;
  clip-path: circle(40px at center);
`;
const UserInfoContainer = styled.div`
  padding: 1.5rem;
  max-width: 100%;
  word-break: break-all;
`;

const AvatarUrl = styled.p`
  font-size: 15px;
  color: white;
  line-height: 20px;
  margin: 0;
  word-break: break-all;
`;

const Login = styled.p`
  font-size: 25px;
  font-weight: bold;
  color: white;
  line-height: 20px;
  margin: 0 0 1rem;
`;

const UserType = styled.p`
  font-size: 20px;
  color: white;
  margin: 0.5rem 0;
`;

export { Container, Avatar, AvatarContainer, AvatarUrl, Login, UserType, UserInfoContainer };
