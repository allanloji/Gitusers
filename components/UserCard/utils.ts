const colors = [
  '#f94144',
  '#F8961E',
  '#F9844A',
  '#F9C74F',
  '#90BE6D',
  '#43AA8B',
  '#4D908E',
  '#577590',
  '#277DA1',
];

export { colors };
