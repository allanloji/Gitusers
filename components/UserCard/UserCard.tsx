import React from 'react';

import {
  Avatar,
  AvatarContainer,
  AvatarUrl,
  Container,
  Login,
  UserInfoContainer,
  UserType,
} from './styles';
import { UserCardProps } from './types';
import { colors } from './utils';

/**
 * UserCard
 */
function UserCard({ avatarUrl, login, type, index }: UserCardProps) {
  const color = index % 9;
  return (
    <Container>
      <AvatarContainer color={colors[color]}>
        <Avatar src={avatarUrl} alt={`Avatar url for ${login}`} />
      </AvatarContainer>
      <UserInfoContainer>
        <Login>{login}</Login>
        {/*<AvatarUrl>{avatarUrl}</AvatarUrl>*/}
        <UserType>{type}</UserType>
      </UserInfoContainer>
    </Container>
  );
}

export default UserCard;
