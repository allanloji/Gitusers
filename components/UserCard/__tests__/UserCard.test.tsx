import React from 'react';
import { render } from '@testing-library/react';

import UserCard from '..';

describe('UserCard', () => {
  test('renders successfully', () => {
    const user = {
      id: 1,
      login: 'test',
      avatar_url: 'https://avatars.githubusercontent.com/u/27031?v=4',
      type: 'user',
      html_url: 'https://github.com/test123',
    };

    const { container } = render(<UserCard user={user} />);

    expect(container).toMatchSnapshot();
  });
});
