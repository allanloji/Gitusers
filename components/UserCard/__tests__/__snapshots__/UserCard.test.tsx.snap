// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renders successfully 1`] = `
<div>
  <a
    href="https://github.com/test123"
    rel="noreferrer noopener"
    target="_blank"
  >
    <div
      class="css-1gi9ab3"
    >
      <div
        class="css-g4ys84"
        color="#F8961E"
      >
        <img
          alt="Avatar url for test"
          class="css-a6g4fg"
          src="https://avatars.githubusercontent.com/u/27031?v=4"
        />
      </div>
      <div
        class="css-1obzys2"
      >
        <p
          class="css-1mceto1"
        >
          test
        </p>
        <p
          class="css-19cwc4a"
        >
          user
        </p>
      </div>
    </div>
  </a>
</div>
`;
